app/models/spree_amazon/address.rb

```diff
@@ -0,0 +1,64 @@
+class SpreeAmazon::Address
+  class << self
+    def find(order_reference)
+      response = mws(order_reference).fetch_order_data
+      from_response(response)
+    end
+
+    def from_response(response)
+      if response.destination["PhysicalDestination"].blank?
+        return nil
+      end
+      new attributes_from_response(response.destination["PhysicalDestination"])
+    end
+
+    private
+
+    def mws(order_reference)
+      AmazonMws.new(order_reference, test_mode)
+    end
```

Comment:
The idea behind the `mws` method is to isolate the dependency on `AmazonMws`.

I'll rename `test_mode` good idea!